File: Java/프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.java
import java.util.*;

class Solution {
    int answer = 0;
    public int solution(int[] ingredient) {
        List<Integer> ingredientList = new ArrayList<Integer>();
        for(int i: ingredient) {
            ingredientList.add(i);
        }
        makeBuger(ingredientList);
        return answer;
    }
    public void makeBuger(List<Integer> list) {
        List<Integer> l1 = list;
        int i;
        int k = l1.size()-3;
        for(i=0; i<k; i++) {
            if(l1.get(i)==1 && l1.get(i+1)==2 && l1.get(i+2)==3 && l1.get(i+3)==1){
                l1.remove(i+3);
                l1.remove(i+2);
                l1.remove(i+1);
                l1.remove(i);
                answer++;
                k-=4;
                if(i-4>=-1) {
                    i-=4;
                }
                else i=-1;
            }
        }
    }
}

File: Kotlin/프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.kt
class Solution {
    fun solution(ingredient: IntArray): Int {
        var answer = 0
        var stack = mutableListOf<Int>()
        for(i in 0..ingredient.lastIndex) {
            stack.add(ingredient[i])
            if(stack.size>=4) {
                if(stack.takeLast(4) == listOf(1,2,3,1)) {
                    repeat(4) { stack.removeAt(stack.lastIndex)}
                    answer++
                }
                
            }
        }
        return answer
    }
}